site/src/pages/CreateTemplatePage/BuildLogsDrawer.tsx

```diff
@@ -0,0 +1,132 @@
+import Drawer from "@mui/material/Drawer";
+import Close from "@mui/icons-material/Close";
+import IconButton from "@mui/material/IconButton";
+import { visuallyHidden } from "@mui/utils";
+import { FC, useEffect, useRef } from "react";
+import { TemplateVersion } from "api/typesGenerated";
+import { WorkspaceBuildLogs } from "modules/workspaces/WorkspaceBuildLogs/WorkspaceBuildLogs";
+import { useTheme } from "@emotion/react";
+import { navHeight } from "theme/constants";
+import { useVersionLogs } from "modules/templates/useVersionLogs";
+import { JobError } from "api/queries/templates";
+import AlertTitle from "@mui/material/AlertTitle";
+import { Alert, AlertDetail } from "components/Alert/Alert";
+import Button from "@mui/material/Button";
+import Collapse from "@mui/material/Collapse";
+
+type BuildLogsDrawerProps = {
+  error: unknown;
+  open: boolean;
+  onClose: () => void;
+  templateVersion: TemplateVersion | undefined;
+  variablesSectionRef: React.RefObject<HTMLDivElement>;
+};
+
+export const BuildLogsDrawer: FC<BuildLogsDrawerProps> = ({
+  templateVersion,
+  error,
+  variablesSectionRef,
+  ...drawerProps
+}) => {
+  const theme = useTheme();
+  const { logs } = useVersionLogs(templateVersion);
+  const logsContainer = useRef<HTMLDivElement>(null);
+
+  const scrollToBottom = () => {
+    setTimeout(() => {
```

Comment:
When dealing with scrolling and focus, I had to use `setTimeout` to make it work but I'm not sure why... maybe because the component was not rendered yet? If yes, how could I solve that in a better way? 🤔

Reply:
you should be calling this from a `useLayoutEffect` rather than `useEffect`

Reply:
I agree, but even with layout effect it needs `setTimeout` to work properly 😢